sympy/solvers/diophantine.py

```diff
@@ -1040,13 +1040,13 @@ def _diop_quadratic(var, coeff, t):
 
         if D < 0:
             for solution in solns_pell:
-                s1 = P*Matrix([solution[0], solution[1]]) + Q
-                s2 = P*Matrix([-solution[0], solution[1]]) + Q
-                try:
-                    sol.add(tuple([as_int(_) for _ in s1]))
-                    sol.add(tuple([as_int(_) for _ in s2]))
-                except ValueError:
-                    pass
+                for X_i in [-solution[0], solution[0]]:
```

Comment:
No, we cannot get all the solutions without this part of code i tried to look into it.
Yes, it is true that it is doing what permute=True does but not in all cases considered the last case mentioned in above comment we are getting only 6 solutions out of 16 even after using permute  = True with original code.